lib/internal/modules/esm/module_job.js

```diff
@@ -73,7 +97,8 @@ class ModuleJob {
       // these `link` callbacks depending on each other.
       const dependencyJobs = [];
       const promises = this.module.link((specifier, assertions) => {
-        const job = this.loader.getModuleJob(specifier, url, assertions);
+        const moduleLoader = require('internal/process/esm_loader').esmLoader;
```

Comment:
Should we cache this line?

Reply:
Why would we need to catch a require?

Reply:
I thought there was a cost of requiring each time? We use a common lazy loading pattern across the repository. Doesn’t this apply here?

Reply:
Oh! Sorry `cache`, not `catch`.

No, we specifically should not: we need a fresh value every time (that's the purpose of this change).